en/middleware/mavlink.md

```diff
@@ -2,200 +2,419 @@
+Note that most generated libraries do not create code for microservices.
+
+The MAVLink project defines a number of [standard messages, commands](https://mavlink.io/en/messages/common.html) and [microservices](https://mavlink.io/en/services/) for exchanging data.
+These are the definitions that most flight stacks, ground stations, and MAVLink peripherals, are likely to support.
+The project also hosts [dialect XML definitions](https://mavlink.io/en/messages/#dialects), which contain MAVLink definitions that are specific to a flight stack or other stakeholder.
+
+The protocol relies on each end of the communication having a shared definition of what messages are being sent.
+What this means is that in order to communicate both ends of the communication must use libraries generated from the same XML definition.
+
+<!--
+The messages are sent over-the-wire in the "payload" of a [MAVLink packet](https://mavlink.io/en/guide/serialization.html#mavlink2_packet_format).
+In order to reduce the amount of information that must be sent, the packet does not include the message metadata, such as what fields are in the message and so on.
+Instead, the fields are serialized in a predefined order based on data size and XML definition order, and MAVLink relies on each end of the communication having a shared definition of what messages are being sent.
+The shared identity of the message is conveyed by the message id, along with a CRC ("`CRC_EXTRA`") that uniquely identifies the message based on its name and id, and the field names and types.
+The receiving end of the communication will discard any packet for which the message id and the `CRC_EXTRA` do not match.
+-->
+
+## PX4 and MAVLink
+
+PX4 uses _standard_ MAVLink definitions by default for the greatest compatibility with MAVLink ground stations, libraries, and external components such as MAVLink cameras.
+These are included from `development.xml` (by default) in the `main` branch.
+
+:::note
+Not all "standard" messages, commands and microservices are _implemented_.
+:::
 
 PX4 includes the [mavlink/mavlink](https://github.com/mavlink/mavlink) repo as a submodule under [/src/modules/mavlink](https://github.com/PX4/PX4-Autopilot/tree/main/src/modules/mavlink), and generates the MAVLink 2 C header files at build time.
 
 There are are number of XML dialect files in [/mavlink/messages/1.0/](https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/).
-The dialect that is built is specified using the variable `MAVLINK_DIALECT` in [/src/modules/mavlink/CMakeLists.txt](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/CMakeLists.txt#L34); by default this is [development.xml](https://github.com/mavlink/mavlink/blob/master/message_definitions/v1.0/development.xml).
+The dialect that is built is defined in the variable `CONFIG_MAVLINK_DIALECT` used in [/src/modules/mavlink/CMakeLists.txt](https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/CMakeLists.txt).
```

Comment:
The value is set in `CONFIG_MAVLINK_DIALECT` rather than `MAVLINK_DIALECT` as indicated, and appears to be set in `default.px4board` for SITL. This seems odd - is the setting there used for ALL boards?

Reply:
The other boards use `common` by default: https://github.com/PX4/PX4-Autopilot/blob/main/src/modules/mavlink/Kconfig#L10

Reply:
Ah. So you'd change `default "common"` there to your XML file if you wanted something custom?

Reply:
You could, but need to be aware if a board has set `CONFIG_MAVLINK_DIALECT`, it will override it.

Reply:
Thanks. The interesting bit here is that common is really the default on real hardware - a bit of a worry since testing is mostly in SITL so if you don't do real hardware testing you could well use definitions that aren't available in the real world.

I made all this a bit more clear in https://github.com/PX4/PX4-user_guide/pull/3027